packages/components/pie-switch/src/index.ts

```diff
@@ -99,9 +102,14 @@ export class PieSwitch extends RtlMixin(LitElement) implements SwitchProps {
     /**
      * The onChange function updates the checkbox state and emits an event for consumers.
      * @param {Event} event - This should be the change event that was listened for on an input element with `type="checkbox"`.
+     * @param {boolean | null} checkedOverride - This is an optional parameter that allows consumers to override the checked state of the switch if they want to manually trigger a change event
      */
-    onChange (event: Event) {
-        const { checked } = event?.currentTarget as HTMLInputElement;
+    onChange (event: Event, checkedOverride: boolean | null = null) {
```

Comment:
I have extended this method to accept a boolean override for the checked state. This allows our label component to explicitly set the checked state if it needs to but still allows this component to be responsible for dispatching it's own change event